### PR TITLE
#trivial Clean up cross domain communicator

### PR DESCRIPTION
**Pull request by bangtoven**

### CODE REVIEW

File: packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts [hunk 1]
```diff
@@ -31,58 +28,17 @@ export abstract class CrossDomainCommunicator {
...
     const targetOrigin = this.getTargetOrigin(options);
     if (!targetOrigin || !this.peerWindow) {
       throw standardErrors.rpc.internal('Communicator: No peer window found');
     }
-
-    const message = createMessage(params);
     this.peerWindow.postMessage(message, targetOrigin);
   }
 
-  async postMessageForResponse<M extends Message>(
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts:47]: FE doesn't use this

File: packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts [hunk 2]
```diff
@@ -31,58 +28,17 @@ export abstract class CrossDomainCommunicator {
     return undefined;
   }
 
-  postMessage<M extends Message>(
-    params: MessageWithOptionalId<M>,
-    options?: { bypassTargetOriginCheck: boolean }
-  ) {
+  async postMessage(message: Message, options?: { bypassTargetOriginCheck: boolean }) {
+    await this.connect();
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts:32]: this change should have no effect. callers of this class used to be responsible to call it. review the code changes below.

File: packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
 export abstract class CrossDomainCommunicator {
   protected url: URL | undefined = undefined;
   private connected = false;
 
-  protected abstract onConnect(): Promise<void>;
-  protected abstract onDisconnect(): void;
-  protected abstract onEvent(event: MessageEvent<Message>): void;
+  protected abstract setupPeerWindow(): Promise<void>;
+  // returns true if the message is handled
+  protected abstract handleIncomingMessage(_: Message): Promise<boolean>;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts:10]: will use this boolean value in the followup pr

**Comment by bangtoven** [comment]
fyi @fan-zhang-sv i am closing this.
let's revisit with our new strategy of not allowing batching.

File: packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts [hunk 4]
```diff
@@ -1,26 +1,25 @@
-import { createMessage, Message, MessageID, MessageWithOptionalId } from '../message';
+import { Message, MessageID } from '../message';
 import { standardErrors } from ':core/error';
 
 export abstract class CrossDomainCommunicator {
   protected url: URL | undefined = undefined;
   private connected = false;
 
-  protected abstract onConnect(): Promise<void>;
-  protected abstract onDisconnect(): void;
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts:9]: FE doesn't use `onDisconnect` nor `disconnect`

File: packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts [hunk 5]
```diff
@@ -1,26 +1,25 @@
...
-  async connect(): Promise<void> {
+  async connect() {
     if (this.connected) return;
     window.addEventListener('message', this.eventListener.bind(this));
-    await this.onConnect();
+    await this.setupPeerWindow();
     this.connected = true;
   }
 
-  disconnect(): void {
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/communicator/CrossDomainCommunicator.ts:19]: FE doesn't use onDisconnect nor disconnect

**Review by fan-zhang-sv — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for fan-zhang-sv @ 2024-05-01 20:17:55 UTC
User failed mfa authentication, public email is not set on your github profile. see [go/mfa-help](http://go/mfa-help)

**Review by fan-zhang-sv — APPROVED** [review]

**Pull request merged**